components/bl-transitions-component/src/collapse-left.js

```diff
@@ -10,26 +10,34 @@ export function CollapseLeft(props) {
   const { onMounted, onUnmounted, onEndAnimation, onStartAnimation } = eventHandlers;
 
   const [width, setWidth] = useState(0);
+  const [initWidth, setInitWidth] = useState('');
 
   const rootRef = useRef();
+  const [element, setElement] = useState({ current: null });
 
-  const setIsAuto = useResizeObserver(rootRef.current, 'width', width, setWidth);
-  const isTransition = useTransition(rootRef, display, duration, width, 'width', setIsAuto, onEndAnimation);
-  const isImagesLoaded = useImageLoad(rootRef, dynamicContent);
+  const isImagesLoaded = useImageLoad(element, dynamicContent);
+  const isTransition = useTransition(element, display, duration, initWidth, width, 'width', onEndAnimation);
+
+  useEffect(() => {
+    if (rootRef.current && !element.current) {
+      setElement({ current: rootRef.current.firstElementChild });
+      setInitWidth(rootRef.current.firstElementChild.style.width);
+    }
+  }, [rootRef]);
 
   useEffect(() => {
     let getWidthTimeout;
 
-    if (rootRef.current && isContentLoaded && isImagesLoaded) {
-      getWidthTimeout = setTimeout(() => {
-        rootRef.current.style.width = 'auto';
-        setWidth(rootRef.current.clientWidth);
-        rootRef.current.style.width = '0px';
-      }, 50);
+    if (element.current && isContentLoaded && isImagesLoaded) {
+      const size = element.current.clientWidth;
+
+      getWidthTimeout = setTimeout(() => setWidth(size), 50);
```

Comment:
Why timeout is `50`?

Reply:
Still unanswered

Reply:
there is a possibility that the component appears already with the final height and the transition is not visible. When I add the timeout to set the final height, I don't see this possibility. It doesn't have to be 50, I put 50 with a margin because for example if you set no delay or less than 5ms you might have this possibility

Reply:
Also I detected this possibility only when component rendered at on time with page

Reply:
What is the reason for this "possibility"?

Each time when you use the timeout with some not reasonable value(50 in your example) it says that we trying to make a workaround in a very imperative way. Why it's bad? For example, I opened the page on a slow device, does 50 still works? Or if render of children's content takes more than 50ms?

Sometimes it makes sense, but at least I want to know we have an understanding of the nature of this behavior.